journal/src/main/java/io/camunda/zeebe/journal/file/SegmentWriter.java

```diff
@@ -260,6 +307,12 @@ public void truncate(final long index) {
       buffer.position(descriptorLength);
       invalidateNextEntry(descriptorLength);
     } else {
+      if (lastEntryPosition > 0) {
```

Comment:
I feel that persisting index is not worth it, because we will delete the segments as fast as we can. So persisting the index just to delete it a few minutes later sounds not useful.